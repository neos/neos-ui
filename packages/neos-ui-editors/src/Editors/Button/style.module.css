.buttonEditor {
    margin-inline: calc(var(--spacing-Quarter) / -2);
}

.buttonEditor .button {
    margin: calc(var(--spacing-Quarter) / 2);
}

.buttonEditor.buttonEditorHighlight {
    box-shadow: 0 0 0 2px var(--colors-Warn);
    border-radius: 2px;
}
