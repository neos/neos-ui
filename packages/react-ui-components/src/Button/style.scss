@import '../reset';

.btn {
    @include reset;
    position: relative;
    display: inline-block;
    height: var(--spacing-GoldenUnit);
    min-width: var(--spacing-GoldenUnit);
    padding: 0 var(--spacing-Full);
    border: 0;
    -webkit-font-smoothing: antialiased;
    vertical-align: middle;
    width: auto;
    color: var(--colors-ContrastBrightest);
    cursor: pointer;
    background-color: var(--colors-ContrastNeutral);
    font-size: var(--fontSize-Base);

    &:focus {
        outline: 1px solid var(--colors-PrimaryBlue);
    }

    &[disabled] {
        opacity: .5;
        cursor: not-allowed;

        &:hover,
        &.btn--isPressed,
        &:hover i,
        i {
            color: var(--colors-ContrastBrightest);
        }
    }
}
.btn--size-small {
    height: 32px;
    font-size: 13px;
    padding: 0 8px;
}

.btn--clean {
    background-color: var(--colors-ContrastDarker);
}
.btn--lighter,
.btn--lighter i {
    background-color: var(--colors-ContrastDark);
}
.btn--neutral,
.btn--neutral i {
    background-color: var(--colors-ContrastNeutral);
}
.btn--transparent {
    background: 0;
}
.btn--brand,
.btn--brand i {
    background-color: var(--colors-PrimaryBlue);
    color: white !important;
}
.btn--success,
.btn--success i {
    background-color: var(--colors-Success);
    color: white !important;
}
.btn--warn,
.btn--warn i {
    background-color: var(--colors-Warn);
}
.btn--error,
.btn--error i {
    background-color: var(--colors-Error);
    color: white !important;
}

.btn--brandActive {
    background-color: var(--colors-PrimaryBlue);

    svg {
        color: var(--colors-ContrastBrightest);
    }
}

.btn--brandHover:hover:enabled,
.btn--brandHover:hover:enabled i,
.btn--brandHover.btn--isPressed,
.btn--brandHover.btn--isPressed i {
    background-color: var(--colors-PrimaryBlue);
}
.btn--cleanHover.btn--isPressed,
.btn--cleanHover.btn--isPressed i,
.btn--cleanHover:hover:enabled,
.btn--cleanHover:hover:enabled i {
    color: var(--colors-PrimaryBlue);

    svg {
        color: var(--colors-PrimaryBlue);
    }
}
.btn--darkenHover.btn--isPressed,
.btn--darkenHover.btn--isPressed i,
.btn--darkenHover:hover:enabled,
.btn--darkenHover:hover:enabled i {
    background-color: rgba(0, 0, 0, .1);
}
.btn--successHover.btn--isPressed,
.btn--successHover.btn--isPressed i,
.btn--successHover:hover:enabled,
.btn--successHover:hover:enabled i {
    background-color: var(--colors-SuccessHover);
}
.btn--warnHover.btn--isPressed,
.btn--warnHover.btn--isPressed i,
.btn--warnHover:hover:enabled,
.btn--warnHover:hover:enabled i {
    background-color: var(--colors-WarnHover);
}
.btn--errorHover:hover:enabled,
.btn--errorHover:hover:enabled i,
.btn--errorHover.btn--isPressed,
.btn--errorHover.btn--isPressed i {
    background-color: var(--colors-ErrorHover);
}
