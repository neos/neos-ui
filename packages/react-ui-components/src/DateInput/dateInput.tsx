import React, {PureComponent} from 'react';
import enhanceWithClickOutside from '../enhanceWithClickOutside/index';
import moment, {Moment, isMoment} from 'moment';
import mergeClassNames from 'classnames';
// @ts-ignore missing typings - we can not update due to a know issue with firefox (https://github.com/neos/neos-ui/pull/1874)
import Collapse from 'react-collapse';
import DatePicker, {TimeConstraints} from 'react-datetime';

import {PickDefaultProps} from '../utils-typescript';
import Icon from '../Icon';

// WHY: Because momentJs locales are not bundled automatically, we have to explicitly add them.
import 'moment/locale/de';

export interface DateInputProps {
    /**
     * The Date instance which represents the selected value.
     */
    readonly value?: Date;

    /**
     * Additional className to render into the wrapper div
     */
    readonly className?: string;

    /**
     * An optional placeholder which will be rendered if no date was selected.
     */
    readonly placeholder?: string;

    /**
     * An optional id for the input.
     */
    readonly id?: string;

    /**
     * The label which will be displayed within the `Select Today` btn.
     */
    readonly todayLabel: string;

    /**
     * The moment format string to use to format the passed value.
     */
    readonly labelFormat?: string;

    /**
     * Display only date picker
     */
    readonly dateOnly?: boolean;

    /**
     * Display only time picker
     */
    readonly timeOnly?: boolean;

    /**
     * Show the Time in 24 hours or 12 hours format
     */
    readonly is24Hour?: boolean;

    /**
     * Add some constraints to the timepicker.
     * It accepts an object with the format { hours: { min: 9, max: 15, step: 2 }},
     * this example means the hours can't be lower than 9 and higher than 15,
     * and it will change adding or subtracting 2 hours everytime the buttons are clicked.
     * The constraints can be added to the hours, minutes, seconds and milliseconds.
     */
    readonly timeConstraints?: TimeConstraints;

    /**
     * Locale for the date picker (determines time format)
     */
    readonly locale: string;

    /**
     * Disable the DateInput
     */
    readonly disabled?: boolean;

    /**
     * The changehandler to call when the date changes.
     */
    readonly onChange: (date: Date | null) => void;

    /**
     * An optional theme using tremr
     */
    readonly theme?: DateInputTheme;
}

interface DateInputTheme {
    'wrapper': string;
    'disabled': string;
    'disabled-cursor': string;
    'calendarInputWrapper': string;
    'calendarIconBtn': string;
    'calendarFakeInputWrapper': string;
    'calendarFakeInputMirror': string;
    'calendarFakeInput': string;
    'applyBtn': string;
    'closeCalendarIconBtn': string;
    'selectTodayBtn': string;
}

const defaultProps: PickDefaultProps<DateInputProps, 'labelFormat' | 'timeConstraints'> = {
    labelFormat: 'DD-MM-YYYY hh:mm A',
    timeConstraints: {
        minutes: {
            min: 0,
            max: 59,
            step: 5
        }
    }
};

interface DateInputState {
    readonly isOpen: boolean

}

const initialState: DateInputState = {
    isOpen: false
};

export class DateInput extends PureComponent<DateInputProps, DateInputState> {
    public static readonly defaultProps = defaultProps;

    public readonly state = initialState;

    public render(): JSX.Element {
        const {
            placeholder,
            theme,
            value,
            className,
            id,
            todayLabel,
            labelFormat,
            dateOnly,
            timeOnly,
            locale,
            disabled
        } = this.props;
        const selectedDate = value ? moment(value).format(labelFormat) : '';

        const wrapper = mergeClassNames(
            theme!.wrapper,
            {
                [theme!.disabled]: disabled
            },
        );

        const calendarInputWrapper = mergeClassNames(className, theme!.calendarInputWrapper);

        const calendarFakeInputMirror = mergeClassNames(
            theme!.calendarFakeInputMirror,
            {
                [theme!['disabled-cursor']]: disabled
            },
        );

        const calendarIconBtn = mergeClassNames(
            theme!.calendarIconBtn,
            {
                [theme!['disabled-cursor']]: disabled
            },
        );

        const closeCalendarIconBtn = mergeClassNames(
            theme!.closeCalendarIconBtn,
            {
                [theme!['disabled-cursor']]: disabled
            },
        );

        return (
            <div className={wrapper}>
                <div className={calendarInputWrapper}>
                    <button
                        onClick={this.handleClick}
                        className={calendarIconBtn}
                    >
                        <Icon icon="far calendar-alt"/>
                    </button>
                    <div className={theme!.calendarFakeInputWrapper}>
                        <div
                            role="presentation"
                            onClick={this.handleClick}
                            className={calendarFakeInputMirror}
                        />
                        <input
                            id={id}
                            onFocus={this.handleFocus}
                            type="datetime"
                            placeholder={placeholder}
                            className={theme!.calendarFakeInput}
                            value={selectedDate}
                            readOnly={true}
                        />
                    </div>
                    <button
                        onClick={this.handleClearValueClick}
                        className={closeCalendarIconBtn}
                    >
                        <Icon icon="times"/>
                    </button>
                </div>
                <Collapse isOpened={this.state.isOpen}>
                    <button
                        className={theme!.selectTodayBtn}
                        onClick={this.handleSelectTodayBtnClick}
                    >
                        {todayLabel}
                    </button>
                    <DatePicker
                        open={true}
                        defaultValue={value}
                        dateFormat={!timeOnly}
                        utc={dateOnly}
                        locale={locale}
                        timeFormat={this.timeFormat}
                        onChange={this.handleChange}

                        timeConstraints={this.props.timeConstraints}
                    />
                </Collapse>
            </div>
        );
    }

    private readonly handleClick = () => {
        if (!this.props.disabled) {
            this.toggle();
        }
    }

    private readonly handleFocus = () => {
        if (!this.props.disabled) {
            this.open();
        }
    }

    private readonly handleClearValueClick = () => {
        if (!this.props.disabled) {
            this.setState({isOpen: false}, () => {
                this.props.onChange(null);
            });
        }
    }

    private readonly handleChange = (value: Moment | string) => {
        const momentVal: Moment = isMoment(value) ? value : moment(value);
        this.props.onChange(momentVal.toDate());
    }

    private readonly handleSelectTodayBtnClick = () => {
        this.setState({
            isOpen: false
        }, () => {
            let date = moment().toDate();
            if (this.props.dateOnly) {
                date = moment().utc().startOf('day').toDate();
            }
            this.props.onChange(date);
        });
    }

    public handleClickOutside = () => this.close();

    private readonly toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    private readonly open = () => {
        this.setState({
            isOpen: true
        });
    }

    private readonly close = () => {
        this.setState({
            isOpen: false
        });
    }

    private get timeFormat(): false | string {
        if (this.props.dateOnly) {
            return false;
        }
        return this.props.is24Hour ? 'HH:m' : 'hh:m A';
    }
}

//
// Add the click-outside functionality to the DateInput component.
//
const EnhancedDateInput = enhanceWithClickOutside(DateInput);

export default EnhancedDateInput;
