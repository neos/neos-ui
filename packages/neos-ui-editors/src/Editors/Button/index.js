import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Icon, IconButton} from '@neos-project/react-ui-components';
import style from './style.module.css';
import cx from 'classnames';
import {neos} from '@neos-project/neos-ui-decorators';
import {connect} from 'react-redux';
import {selectors} from '@neos-project/neos-ui-redux-store';

const getDataLoaderOptionsForProps = props => ({
    contextNodePath: props.focusedNodePath,
    dataSourceIdentifier: props.options.dataSourceIdentifier,
    dataSourceUri: props.options.dataSourceUri,
    dataSourceAdditionalData: props.options.dataSourceAdditionalData,
    dataSourceDisableCaching: Boolean(props.options.dataSourceDisableCaching)
});

@neos(globalRegistry => ({
    i18nRegistry: globalRegistry.get('i18n'),
    dataSourcesDataLoader: globalRegistry.get('dataLoaders').get('DataSources')
}))
@connect(state => ({
    focusedNodePath: selectors.CR.Nodes.focusedNodePathSelector(state)
}))

export default class ButtonEditor extends Component {
    static propTypes = {
        commit: PropTypes.func.isRequired,
        highlight: PropTypes.bool,
        value: PropTypes.any,
        className: PropTypes.string,
        options: PropTypes.shape({
            allowEmpty: PropTypes.bool,
            multiple: PropTypes.bool,
            disabled: PropTypes.bool,
            values: PropTypes.objectOf(PropTypes.shape({
                label: PropTypes.string,
                icon: PropTypes.string,
                iconActive: PropTypes.string,
                disabled: PropTypes.bool
            })),

            dataSourceIdentifier: PropTypes.string,
            dataSourceUri: PropTypes.string,
            dataSourceDisableCaching: PropTypes.bool,
            dataSourceAdditionalData: PropTypes.objectOf(PropTypes.any)
        }).isRequired,

        i18nRegistry: PropTypes.object.isRequired
    };

    static defaultOptions = {
        allowEmpty: true,
        multiple: false,
        disabled: false
    };

    state = {
        selected: [],
        isLoading: false,
        buttons: []
    };

    options = {};

    initialValueType = 'string';

    constructor(props) {
        super(props);
        const {commit, options, value} = props;
        this.options = Object.assign({}, this.constructor.defaultOptions, options)
        // TODO: get actual property data type
        this.initialValueType = Array.isArray(value) ? 'array' : 'string';
        if (this.options.multiple && this.initialValueType === 'string') {
            console.warn(`Misconfiguration in property "${props.identifier}". Multiple is activated but value type seems to be "string" but should be "array".`);
        }

        this.state = {
            selected: Array.isArray(value) ? value : (value ? [value] : []),
            buttons: this.hasDataSource() ? [] : this.createButtonsFromConfiguration(this.options.values)
        };
        this.commit = commit;
    }

    hasDataSource() {
        return this.options.dataSourceIdentifier || this.options.dataSourceUri;
    }

    componentDidMount() {
        if (this.hasDataSource()) {
            this.loadDataSourceOptions();
        }
    }

    loadDataSourceOptions() {
        this.setState({isLoading: true});
        this.props.dataSourcesDataLoader.resolveValue(getDataLoaderOptionsForProps(this.props), this.props.value)
            .then(configuredValues => {
                this.setState({
                    isLoading: false,
                    buttons: this.createButtonsFromConfiguration(configuredValues)
                });
            });
    }

    createButtonsFromConfiguration(configuredValues) {
        return Object.entries(configuredValues).map(([key, configuredValue]) => {
            return {
                value: configuredValue.value ?? key,
                label: this.props.i18nRegistry.translate(configuredValue.label),
                icon: configuredValue.icon,
                iconActive: configuredValue.iconActive
            };
        });
    }

    handleSelect(selectedValue) {
        const {multiple, allowEmpty} = this.options;
        // we'd like to have a copy instead of ref
        let selected = [...this.state.selected];
        if (multiple === false) {
            if (selected.includes(selectedValue)) {
                selected = [];
            } else {
                selected = [selectedValue];
            }
        } else if (selected.includes(selectedValue)) {
            selected.splice(selected.indexOf(selectedValue), 1);
        } else {
            selected.push(selectedValue);
        }

        // if allowEmpty is false but new selected length will be 0, drop changes
        if (!allowEmpty && selected.length === 0) {
            return;
        }
        this.setState({selected});

        this.commit(this.initialValueType === 'string' ? selected[0] || '' : selected);
    }

    isSelected(val) {
        return this.state.selected.includes(val);
    }

    render() {
        return (<div className={cx(style.buttonEditor, this.props.highlight && style.buttonEditorHighlight)}>
            {
                this.state.isLoading
                    ? <Icon icon="spinner" size="lg" spin/>
                    : this.state.buttons.map((button) => (<IconButton
                            key={button.value}
                            style="lighter"
                            hoverStyle="darken"
                            className={style.button}
                            icon={button.iconActive ? (this.isSelected(button.value) ? button.iconActive : button.icon) : button.icon}
                            title={button.label}
                            disabled={this.options.disabled || button.disabled}
                            isActive={this.isSelected(button.value)}
                            onClick={this.handleSelect.bind(this, button.value)}
                    />))
            }
        </div>);
    }
}
