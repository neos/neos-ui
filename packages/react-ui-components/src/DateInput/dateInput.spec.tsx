import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import moment from 'moment';
import DatePicker from 'react-datetime';

import Button from '../Button';
import {DateInput} from './dateInput';

describe('<DateInput/>', () => {
    const props: DateInput['props'] = {
        onChange: jest.fn(),
        locale: 'en-US',
        applyLabel: 'applyLabel',
        todayLabel: 'todayLabel',
        theme: {
            'wrapper': 'wrapperClassName',
            'disabled': 'disabledClassName',
            'disabled-cursor': 'disabledClassName',
            'calendarInputWrapper': 'calendarInputWrapperClassName',
            'calendarIconBtn': 'calendarIconBtnClassName',
            'calendarFakeInputWrapper': 'calendarFakeInputWrapperClassName',
            'calendarFakeInputMirror': 'calendarFakeInputMirrorClassName',
            'calendarFakeInput': 'calendarFakeInputClassName',
            'applyBtn': 'applyBtnClassName',
            'closeCalendarIconBtn': 'closeCalendarIconBtnClassName',
            'selectTodayBtn': 'selectTodayBtnClassName',
        }
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<DateInput {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should initialize with a state of {isOpen = false}.', () => {
        const wrapper = shallow(<DateInput {...props}/>);

        expect(wrapper.state('isOpen')).toBe(false);
    });

    it('should set the "isOpen" state to a truthy value when clicking on the first button.', () => {
        const wrapper = shallow(<DateInput {...props}/>);
        const btn = wrapper.find('button').at(0);

        btn.simulate('click');

        expect(wrapper.state('isOpen')).toBe(true);
    });

    it('should set the "isOpen" state to a falsy value and call the "onChange" prop with `null` when clicking on the second button.', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<DateInput {...props} onChange={onChange}/>);
        const btn = wrapper.find('button').at(1);

        wrapper.setState({isOpen: true});
        btn.simulate('click');

        expect(wrapper.state('isOpen')).toBe(false);
        expect(onChange.mock.calls.length).toBe(1);
        expect(onChange.mock.calls[0][0]).toBe(null);
    });

    it('should call the "onChange" prop when triggering the change event on the "DatePicker" Component and clicking apply.', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<DateInput {...props} value={new Date()} onChange={onChange}/>);
        const picker = wrapper.find(DatePicker);
        const applyButton = wrapper.find(Button);
        const newVal = moment();

        picker.simulate('change', newVal);
        applyButton.simulate('click');

        expect(onChange.mock.calls.length).toBe(1);
        expect(onChange.mock.calls[0][0].toTimeString()).toBe(newVal.toDate().toTimeString());
    });

    it('should set the "isOpen" state to a falsy value and call the "onChange" prop with todays date when clicking on the third button.', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<DateInput {...props} onChange={onChange}/>);
        const btn = wrapper.find('button').at(2);
        const date = new Date();

        wrapper.setState({isOpen: true});
        btn.simulate('click');

        expect(wrapper.state('isOpen')).toBe(false);
        expect(onChange.mock.calls.length).toBe(1);
        expect(onChange.mock.calls[0][0].toTimeString()).toBe(date.toTimeString());
    });

    it('should not set "UTC" on DatePickerComponent if "dateOnly" prop is set.', () => {
        const wrapper = shallow(<DateInput {...props} dateOnly={true}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call "onChange" with beginning of day in UTC for todays date if "dateOnly" is set.', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<DateInput {...props} dateOnly={true} onChange={onChange}/>);
        const btn = wrapper.find('button').at(2);
        const date = new Date();

        wrapper.setState({isOpen: true});
        btn.simulate('click');

        expect(onChange.mock.calls.length).toBe(1);

        const receivedDate = onChange.mock.calls[0][0];
        expect(receivedDate.getUTCDate()).toBe(date.getDate());
        expect(receivedDate.getUTCMonth()).toBe(date.getMonth());
        expect(receivedDate.getUTCFullYear()).toBe(date.getFullYear());
        expect(receivedDate.getUTCHours()).toBe(0);
        expect(receivedDate.getUTCMinutes()).toBe(0);
        expect(receivedDate.getUTCSeconds()).toBe(0);
        expect(receivedDate.getUTCMilliseconds()).toBe(0);
    });
});
