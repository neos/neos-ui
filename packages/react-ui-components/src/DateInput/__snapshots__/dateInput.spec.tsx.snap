// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInput/> should render correctly. 1`] = `
<div
  className="wrapperClassName"
>
  <div
    className="calendarInputWrapperClassName"
  >
    <button
      className="calendarIconBtnClassName"
      onClick={[Function]}
    >
      <ThemedIcon
        color="default"
        composeTheme="deeply"
        icon="far calendar-alt"
        mapThemrProps={[Function]}
        padded="none"
        size="sm"
      />
    </button>
    <div
      className="calendarFakeInputWrapperClassName"
    >
      <div
        className="calendarFakeInputMirrorClassName"
        onClick={[Function]}
        role="presentation"
      />
      <input
        className="calendarFakeInputClassName"
        onFocus={[Function]}
        readOnly={true}
        type="datetime"
        value=""
      />
    </div>
    <button
      className="closeCalendarIconBtnClassName"
      onClick={[Function]}
    >
      <ThemedIcon
        color="default"
        composeTheme="deeply"
        icon="times"
        mapThemrProps={[Function]}
        padded="none"
        size="sm"
      />
    </button>
  </div>
  <UnmountClosed
    isOpened={false}
  >
    <button
      className="selectTodayBtnClassName"
      onClick={[Function]}
    >
      todayLabel
    </button>
    <DateTime
      className=""
      closeOnSelect={false}
      closeOnTab={true}
      dateFormat={true}
      defaultValue=""
      input={true}
      inputProps={Object {}}
      locale="en-US"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onNavigateBack={[Function]}
      onNavigateForward={[Function]}
      onViewModeChange={[Function]}
      open={true}
      strictParsing={true}
      timeConstraints={
        Object {
          "minutes": Object {
            "max": 59,
            "min": 0,
            "step": 5,
          },
        }
      }
      timeFormat={true}
      utc={false}
    />
    <ThemedButton
      _refHandler={[Function]}
      className="applyBtnClassName"
      composeTheme="deeply"
      disabled={false}
      hoverStyle="brand"
      isActive={false}
      isFocused={false}
      mapThemrProps={[Function]}
      onClick={[Function]}
      size="regular"
      style="brand"
      type="button"
    >
      applyLabel
    </ThemedButton>
  </UnmountClosed>
</div>
`;

exports[`<DateInput/> should set "utc" on DatePickerComponent if "dateOnly" prop is set. 1`] = `
<div
  className="wrapperClassName"
>
  <div
    className="calendarInputWrapperClassName"
  >
    <button
      className="calendarIconBtnClassName"
      onClick={[Function]}
    >
      <ThemedIcon
        color="default"
        composeTheme="deeply"
        icon="far calendar-alt"
        mapThemrProps={[Function]}
        padded="none"
        size="sm"
      />
    </button>
    <div
      className="calendarFakeInputWrapperClassName"
    >
      <div
        className="calendarFakeInputMirrorClassName"
        onClick={[Function]}
        role="presentation"
      />
      <input
        className="calendarFakeInputClassName"
        onFocus={[Function]}
        readOnly={true}
        type="datetime"
        value=""
      />
    </div>
    <button
      className="closeCalendarIconBtnClassName"
      onClick={[Function]}
    >
      <ThemedIcon
        color="default"
        composeTheme="deeply"
        icon="times"
        mapThemrProps={[Function]}
        padded="none"
        size="sm"
      />
    </button>
  </div>
  <UnmountClosed
    isOpened={false}
  >
    <button
      className="selectTodayBtnClassName"
      onClick={[Function]}
    >
      todayLabel
    </button>
    <DateTime
      className=""
      closeOnSelect={false}
      closeOnTab={true}
      dateFormat={true}
      defaultValue=""
      input={true}
      inputProps={Object {}}
      locale="en-US"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onNavigateBack={[Function]}
      onNavigateForward={[Function]}
      onViewModeChange={[Function]}
      open={true}
      strictParsing={true}
      timeConstraints={
        Object {
          "minutes": Object {
            "max": 59,
            "min": 0,
            "step": 5,
          },
        }
      }
      timeFormat={false}
      utc={true}
    />
    <ThemedButton
      _refHandler={[Function]}
      className="applyBtnClassName"
      composeTheme="deeply"
      disabled={false}
      hoverStyle="brand"
      isActive={false}
      isFocused={false}
      mapThemrProps={[Function]}
      onClick={[Function]}
      size="regular"
      style="brand"
      type="button"
    >
      applyLabel
    </ThemedButton>
  </UnmountClosed>
</div>
`;
